backend/src/packages/projects/project.repository.ts

```diff
@@ -17,11 +17,19 @@ class ProjectRepository
     return projects.map((project) => ProjectEntity.initialize(project));
   }
 
-  public async findByUserId(id: number): Promise<ProjectEntity[]> {
-    const projects = await this.projectModel
-      .query()
-      .where('userId', id)
-      .execute();
+  public async findByUserId(
+    query: string,
+    id: number,
+  ): Promise<ProjectEntity[]> {
+    let projectQuery = this.projectModel.query().where('userId', id);
+
+    if (query) {
+      projectQuery = projectQuery.whereRaw('LOWER(name) LIKE ?', [
+        `%${query.toLowerCase()}%`,
```

Comment:
In case if filtering options would be increased (there would be filtering not only by name, but also for other fields) - then you'd change the whole implementation, so it would be great to provide the implementation that could be enhanced (you can take request query string params as objects and parse them to construct proper filtering clause)
+ instead of lowercasing everything you can just use `ILIKE`
+instead of mutating builded query you can use functionality provided by knex, to pass conditional clauses eg:
```
this.projectModel.query()
    .where('userId', id)
    .andWhere(getWhereNameClause(nameToFilterBy))
    
    ....
    
    const getWhereNameClause = (name: string | undefined) => (builder: QueryBuilder): void => {
        if (!name) {
            return;
        }
        builder.whereRaw(<your condition>);
    }
```